hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/IMapSqlConnector.java

```diff
@@ -169,7 +173,35 @@ public Table createTable(
                 valueMetadata.getUpsertTargetDescriptor(),
                 indexes,
                 hd,
-                partitioningAttributes);
+                partitioningAttributes,
+                supportsPartitionPruning(nodeEngine, mapName));
+    }
+
+    private boolean supportsPartitionPruning(final NodeEngine nodeEngine, final String mapName) {
+        final MapConfig mapConfig = nodeEngine.getConfig().getMapConfig(mapName);
```

Comment:
if I understand correctly, this entire method boils down to: `return partitioningStrategy instaceof DefaultPartitionStrategy || partitioningStrategy instanceof AttributePartitionStrategy`. What is the reason to do it in a more verbose way? Do you want not to touch `MapContainer`?

Reply:
MapContainer might not exist if the IMap is empty (don't if if there are other cases). This actually could influence plan caching. However, in other places we use MapContainer (eg. for indexes).